### ISSUE TITLE
AppImage crash on arch linux

### ISSUE THREAD

**Issue by taryujiai** [tagspaces/tagspaces#2014]
**Describe the bug**
Windows dissapears. 

**To Reproduce**
1. Download tagspaces-linux-x86_64-5.6.2.AppImage
2. chmod 755 tagspaces-linux-x86_64-5.6.2.AppImage
3. ./tagspaces-linux-x86_64-5.6.2.AppImage
4. Lots of JS output...
At the end:

Error: Error: write EPIPE
    at ChildProcess._send (node:internal/child_process:865:20)
    at ChildProcess.send (node:internal/child_process:738:19)
    at stop (/tmp/.mount_tagspaIm3IMJ/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:215:40)
    at forEach (/tmp/.mount_tagspaIm3IMJ/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:40)
    at Array.forEach (<anonymous>)
    at Object.pm2 [as stopAll] (/tmp/.mount_tagspaIm3IMJ/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:26)
    at App.<anonymous> (/tmp/.mount_tagspaIm3IMJ/resources/app.asar/webpack:/tagspaces/app/main.dev.ts:573:3)
    at App.emit (node:events:525:35)
    at process.<anonymous> (/tmp/.mount_tagspaIm3IMJ/resources/app.asar/webpack:/tagspaces/app/main.dev.ts:774:13)
    at process.apply [as emit] (node:events:525:35)
    at process.on.process.emit (/tmp/.mount_tagspaIm3IMJ/resources/app.asar/webpack:/tagspaces/node_modules/source-map-support/source-map-support.js:516:21)
    at process._fatalException (node:internal/process/execution:149:25)

Node.js v18.16.1

**Expected behavior**
Stay open and functional. 5.5.4 works just fine.

**Screenshots**
N/A

**Desktop Application:**
 - Operating System: Arch Linux
 - TagSpaces Version 5.6.2

**Additional context**
N/A

**Comment by taryujiai** [comment]
$ md5sum tagspaces-linux-x86_64-5.6.2.AppImage 
42af73482d511e18db7ea34d4aca3805  tagspaces-linux-x86_64-5.6.2.AppImage

$ ls -al tagspaces-linux-x86_64-5.6.2.AppImage 
-rwxr-x--- 1 user user 143210996 Dec 11 13:31 tagspaces-linux-x86_64-5.6.2.AppImage

$ yay -Q | grep fuse
confuse 3.3-3
fuse-common 3.16.2-1
fuse2 2.9.9-4
fuse3 3.16.2-1
squashfuse 0.5.0-1

I haven't created any AppImages but my understanding is they are supposed to be self contained and work anywhere so I'm a bit confused by the crash.

I just downloaded it again and launched it. It was up a few seconds 

Spit this out:
./tagspaces-linux-x86_64-5.6.2.AppImage 
[2722:1211/133611.015768:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.DBus.Properties.Get: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.FileChooser”
[2722:1211/133611.015810:ERROR:select_file_dialog_linux_portal.cc(274)] Failed to read portal version property
[2722:1211/133611.156373:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.portal.Settings.Read: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Settings” on object at path /org/freedesktop/portal/desktop
Starting Tagspaces WS on /tmp/.mount_tagspaemwNBs/resources - pid (2786)

---------------------> this showed up a few seconds after above and the app seemed OK but not functional.

[2762:1211/133616.137250:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!

--------------------> Crash and a bunch of what looks like JS (it fills the buffer and more). Here's the end of it. Maybe it will help?

"))),e.ipcMain.on("focus-window",(()=>{ze&&ze.focus()})),e.ipcMain.handle("get-device-paths",(()=>{const a={desktopFolder:e.app.getPath("desktop"),documentsFolder:e.app.getPath("documents"),downloadsFolder:e.app.getPath("downloads"),musicFolder:e.app.getPath("music"),picturesFolder:e.app.getPath("pictures"),videosFolder:e.app.getPath("videos")};return Me&&(a.iCloudFolder=e.app.getPath("home")+"/Library/Mobile Documents/com~apple~CloudDocs"),a})),e.ipcMain.on("get-user-home-path",(a=>{a.returnValue=e.app.getPath("home")})),e.ipcMain.on("worker-response",((e,a)=>{ze&&ze.webContents.send(a.id,a)})),e.ipcMain.handle("select-directory-dialog",(async()=>{const a=await e.dialog.showOpenDialog({properties:["openDirectory","createDirectory"]});return!(!a.filePaths||!a.filePaths.length)&&a.filePaths})),e.ipcMain.on("app-data-path-request",(a=>{a.returnValue=e.app.getPath("appData")})),e.ipcMain.on("app-version-request",(a=>{a.returnValue=e.app.getVersion()})),e.ipcMain.handle("move-to-trash",(async(a,t)=>{const i=[];let n;t.forEach((a=>{i.push(e.shell.trashItem(a))}));try{n=await Promise.all(i)}catch(e){console.error("moveToTrash "+JSON.stringify(t)+"error:",e)}return n})),e.ipcMain.on("set-language",((e,t)=>{a.changeLanguage(t)})),e.ipcMain.on("global-shortcuts-enabled",((a,t)=>{t?(e.globalShortcut.register("CommandOrControl+Shift+F",da),e.globalShortcut.register("CommandOrControl+Shift+P",va),e.globalShortcut.register("MediaPlayPause",va),e.globalShortcut.register("CommandOrControl+Shift+N",ca),e.globalShortcut.register("CommandOrControl+Shift+D",ua),e.globalShortcut.register("MediaNextTrack",ua),e.globalShortcut.register("CommandOrControl+Shift+A",ma),e.globalShortcut.register("MediaPreviousTrack",ma),e.globalShortcut.register("CommandOrControl+Shift+W",He)):e.globalShortcut.unregisterAll()})),e.ipcMain.on("relaunch-app",ga),e.ipcMain.on("quit-application",(()=>{e.globalShortcut.unregisterAll(),e.app.quit()})),process.on("uncaughtException",(e=>{if(e.stack)throw console.error("error:",e.stack),new Error(e.stack);ga()}))}))})(),module.exports=r})();

Error: Error: write EPIPE
    at ChildProcess._send (node:internal/child_process:865:20)
    at ChildProcess.send (node:internal/child_process:738:19)
    at stop (/tmp/.mount_tagspadWaAQi/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:215:40)
    at forEach (/tmp/.mount_tagspadWaAQi/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:40)
    at Array.forEach (<anonymous>)
    at Object.pm2 [as stopAll] (/tmp/.mount_tagspadWaAQi/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:26)
    at App.<anonymous> (/tmp/.mount_tagspadWaAQi/resources/app.asar/webpack:/tagspaces/app/main.dev.ts:573:3)
    at App.emit (node:events:525:35)
    at process.<anonymous> (/tmp/.mount_tagspadWaAQi/resources/app.asar/webpack:/tagspaces/app/main.dev.ts:774:13)
    at process.apply [as emit] (node:events:525:35)
    at process.on.process.emit (/tmp/.mount_tagspadWaAQi/resources/app.asar/webpack:/tagspaces/node_modules/source-map-support/source-map-support.js:516:21)
    at process._fatalException (node:internal/process/execution:149:25)

Node.js v18.16.1

**Comment by taryujiai** [comment]
Just tried the tar.gz

Same error. Let me know if you want me to post the output but it's pretty much the same sequence of events with write EPIPE. I feel like it would just clutter the post.

So I run a bunch of other AppImages (including tagspaces 5.5.4) with no issues.

Here's the output from 5.5.4
$ ./tagspaces-linux-x86_64-5.5.4.AppImage 
[7557:1213/062710.023392:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.DBus.Properties.Get: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.FileChooser”
[7557:1213/062710.023440:ERROR:select_file_dialog_linux_portal.cc(274)] Failed to read portal version property
Starting Tagspaces WS on /tmp/.mount_tagspaBr4jKD/resources - pid (7623)

------------------> Same behavior here. Doesn't show at launch but starts to emit after having the GUI up for a bit.

[7599:1213/062715.181894:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[7599:1213/062718.969549:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[7599:1213/062718.983940:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!

**Comment by sytolk** [comment]
@taryujiai it looks like native Linux error. What is you distribution? and desktop GNOME KDE GTK?
do you have linux packages like: `xdg-desktop-portal-gnome` or `xdg-desktop-portal-gtk` or `xdg-desktop-portal-kde`
can you [pastebin](https://pastebin.com/) the whole error log

**Comment by taryujiai** [comment]
Hi All: I did more troubleshooting.

I did not have either of those xdg packages installed so I installed them. I tested on multiple Arch machines and the behavior is the same for both the AppImage and the tar.gz release.

The log is too big for pastebin: ~1M in size. But it boils down to these entries (very similar for both):
Starting Tagspaces WS on /tmp/.mount_tagspaqbLKQT/resources - pid (2352)
error: Error: write EPIPE
    at ChildProcess._send (node:internal/child_process:865:20)
    at ChildProcess.send (node:internal/child_process:738:19)
    at stop (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:215:40)
    at forEach (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:40)
    at Array.forEach (<anonymous>)
    at Object.pm2 [as stopAll] (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:26)
    at App.<anonymous> (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/app/main.dev.ts:573:3)
    at App.emit (node:events:525:35)
/tmp/.mount_tagspaqbLKQT/resources/app.asar/main.prod.js:2

------------->

Looks like the source file for some JS (largest part by far - it's all one line - minified JS).

------------->

Then what I've posted before:

rror: Error: write EPIPE
    at ChildProcess._send (node:internal/child_process:865:20)
    at ChildProcess.send (node:internal/child_process:738:19)
    at stop (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:215:40)
    at forEach (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:40)
    at Array.forEach (<anonymous>)
    at Object.pm2 [as stopAll] (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:26)
    at App.<anonymous> (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/app/main.dev.ts:573:3)
    at App.emit (node:events:525:35)
    at process.<anonymous> (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/app/main.dev.ts:774:13)
    at process.apply [as emit] (node:events:525:35)
    at process.on.process.emit (/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/node_modules/source-map-support/source-map-support.js:516:21)
    at process._fatalException (node:internal/process/execution:149:25)

Node.js v18.16.1

More on the behavior.

The windows will stay open and load just fine (Locations are there, etc).

It's not until I roll the mouse cursor over the TagSpaces window does it crash and disappear. Not even clicking in to focus the TagSpaces window, just need to rollover it.  

I've updated the title to indicate it's not an AppImage Specific issue.

So to summarize:
This appears in stdout:
./tagspaces-linux-x86_64-5.6.2.AppImage 
Starting Tagspaces WS on /tmp/.mount_tagspaTCxGDl/resources - pid (23039)
[22993:1214/084426.462931:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!

Nothing happens until I roll the mouse over the TagSpaces window without even a click to focus the Window. The terminal still has the mouse focus during the crash.

Then the rest of the output gets dumped. EPIPE, minified JS, etc. and the window goes away.

**Comment by sytolk** [comment]
@taryujiai what is your Linux Desktop? in terminal `echo $XDG_CURRENT_DESKTOP`
This error from [PM2 process](https://github.com/Unitech/pm2) here means only that TagSpaces Electron main thread is not alive 
`(/tmp/.mount_tagspaqbLKQT/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:215:40)`
but the real reason for crash is in main thread and the log is before this log. [Pastebin limit is 50MB](https://www.quora.com/What-is-the-file-size-limit-for-Pastebin) just submit the whole log in pastebin form and copy link here. Better use [debugtron](https://github.com/pd4d10/debugtron) to start TagSpaces and copy log -> after installing debugtron and tagspaces in you system tagspaces will appears in app toolbar of debugtron.

**Comment by curious-energy** [comment]
In my Ubuntu 20 LTS, this problem is also exist in AppImage and deb.

Here is the Error messages:
```
Error: Error: write EPIPE
    at ChildProcess._send (node:internal/child_process:865:20)
    at ChildProcess.send (node:internal/child_process:738:19)
    at stop (/tmp/.mount_tagspaEhCEqz/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:215:40)
    at forEach (/tmp/.mount_tagspaEhCEqz/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:40)
    at Array.forEach (<anonymous>)
    at Object.pm2 [as stopAll] (/tmp/.mount_tagspaEhCEqz/resources/app.asar/webpack:/tagspaces/node_modules/@elife/pm2/index.js:151:26)
    at App.<anonymous> (/tmp/.mount_tagspaEhCEqz/resources/app.asar/webpack:/tagspaces/app/main.dev.ts:573:3)
    at App.emit (node:events:525:35)
    at process.<anonymous> (/tmp/.mount_tagspaEhCEqz/resources/app.asar/webpack:/tagspaces/app/main.dev.ts:774:13)
    at process.apply [as emit] (node:events:525:35)
    at process.on.process.emit (/tmp/.mount_tagspaEhCEqz/resources/app.asar/webpack:/tagspaces/node_modules/source-map-support/source-map-support.js:516:21)
    at process._fatalException (node:internal/process/execution:149:25)

Node.js v18.16.1
```

**Comment by taryujiai** [comment]
5.7.1 AppImage / tag.gz seems to fix the issue. Thank you!
